modules/tinymce/src/core/main/ts/fmt/CaretFormat.ts

```diff
@@ -270,18 +270,21 @@ const removeCaretFormat = (editor: Editor, name: string, vars?: FormatVars, simi
   }
 
   const parents: Node[] = [];
+  let noMatch = true;
   let formatNode: Element | undefined;
-  while (node) {
+  while (node && isFormatElement(editor, SugarElement.fromDom(node))) {
     if (MatchFormat.matchNode(editor, node, name, vars, similar)) {
-      formatNode = node as Element;
-      break;
-    }
-
-    if (node.nextSibling) {
-      hasContentAfter = true;
+      if (noMatch) {
+        formatNode = node as Element;
+        noMatch = false;
+      }
+    } else {
```

Comment:
You could move the `noMatch` condition to before the `MatchFormat.matchNode` call and then it wouldn't run once a match is found.

Reply:
nah that won't work when the format are nested like this
```<p><span style="text-decoration: underline;"><span style="color: red; text-decoration: underline;"><br></span></span></p>```
when toggling underline, `noMatch` will be set to false once the first `span` is matched, then the second `underline` will get pushed into `parents`then concat to the new caret.

Reply:
I just mean this condition doesn't do anything when noMatch is false, so why bother with the call to matchNode first if you can skip it by including noMatch before it.

Reply:
gonna split this into a different ticket, accidentally out of scoop